DS13/code/modules/overmap/transporter.dm

```diff
@@ -145,6 +190,28 @@
 		T.retrieve(thehewmon)
 	playsound(loc, 'DS13/sound/effects/transporter/retrieve.ogg', 100, 4)
 
+/obj/machinery/computer/camera_advanced/transporter_control/proc/transporters_lock()
+	var/list/mobs = list() //mobs near the eyeobj.
+	for(var/mob/living/thehewmon in orange(eyeobj,3))
+		if(isliving(thehewmon))
+			mobs += thehewmon
+	if(!mobs.len)
+		to_chat(operator, "There is no one to lock onto!")
+		return
+	var/question = alert("Lock onto a specific person? or everyone nearby?",name,"Specific person","Nearby")
+	if(question == "Specific person" || !question)
+		var/A
+		A = input("To whom shall we lock on?", "Transporter pattern buffer", A) as null|anything in mobs//overmap_objects
+		if(!A)
+			return
+		locked_on += A
```

Comment:
What if `A` is invalid?

> Operator locked onto <b>deez nuts!</b>.

Reply:
A can only be an atom (object class). Earlier on we also checked that any entry into this list is a mob, so it's impossible to lock onto a text string or something.